### PR TITLE
Allow plugins to define new settings.

### PR DESCRIPTION
**Pull request by shlevy**

### CODE REVIEW

**Comment by shlevy** [comment]
(example use case is implementation of extra builtins that allows the user to specify the path to the extra builtins file)

**Comment by shlevy** [comment]
@edolstra thoughts?

**Comment by puffnfresh** [comment]
Related is https://github.com/NixOS/nix/issues/1875 - maybe this PR fixes it.

**Comment by shlevy** [comment]
Hmm not sure if that's related?

File: src/libutil/config.cc [hunk 1]
```diff
@@ -21,34 +23,37 @@ void Config::addSetting(AbstractSetting * setting)
...
-                initials.erase(i);
+                extras.erase(i);
                 set = true;
             }
         }
     }
 }
 
-void Config::warnUnknownSettings()
+void Config::handleUnknownSettings(bool fatal)
```
Comment (puffnfresh) [inline src/libutil/config.cc]: Is this called with `fatal = true` anywhere now?

Comment (shlevy) [reply src/libutil/config.cc]: @puffnfresh It wasn't before this commit either, but @edolstra removed the flag ed73d40c3b19dc0581bbf28ef29aad50cab3aaf2

**Pull request merged**